src/golang/lib/workflow/operator/operator.go

```diff
@@ -114,20 +115,25 @@ func NewOperator(
 		metadataPath = outputExecPaths[0].OpMetadataPath
 	}
 
-	jobConfig, err := GenerateJobManagerConfig(
-		ctx,
-		opEngineConfig,
-		storageConfig,
-		aqPath,
-		vaultObject,
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "Unable to generate JobManagerConfig.")
-	}
-
-	jobManager, err := job.NewJobManager(jobConfig)
-	if err != nil {
-		return nil, errors.Wrap(err, "Unable to create JobManager.")
+	var jobManager job.JobManager
+	if engineJobManager == nil {
+		jobConfig, err := GenerateJobManagerConfig(
+			ctx,
+			opEngineConfig,
+			storageConfig,
+			aqPath,
+			vaultObject,
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to generate JobManagerConfig.")
+		}
+
+		jobManager, err = job.NewJobManager(jobConfig)
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to create JobManager.")
```

Comment:
This code is duplicated multiple times. Can we make the creation of a new job manager from the fields a helper in `lib/job/`?